### PR TITLE
WIP #3727: Add assertions to help diagnose error

### PR DESCRIPTION
**Pull request by e-carlin**
From looking through the logs it is unclear what exactly caused the error.
These assertions and logging will hopefully narrow down places where we are in a
state we do not expect (ex beleive we are in a request but flask.request is
False).

### CODE REVIEW

File: sirepo/util.py [hunk 1]
```diff
@@ -185,7 +187,14 @@ def flask_app():
 
 def in_flask_request():
     import sys
```
Comment (robnagler) [inline sirepo/util.py:189]: Move this globally.

File: sirepo/util.py [hunk 2]
```diff
@@ -185,7 +187,14 @@ def flask_app():
...
+    global _logged_in_request, _logged_outside_request
     f = sys.modules.get('flask')
+    if _logged_in_request < 10 and f and f.request:
```
Comment (robnagler) [inline sirepo/util.py:192]: I think it is important to note that this is a global that will be accessed in a threaded context. It's ok, but it just should be noted that the number "10" doesn't need to be exact.

I would this code this way:

```py
if not f:
    if _log_not_flask < 10:
        _log_not_flask += 1
        pkdlog('flask is not imported')
    return False
if not f.request:
    if _log_not_request < 10:
        _log_not_request += 1
        pkdlog('flask.request is False')
    return False
return True
```

**Review by robnagler — CHANGES_REQUESTED** [review]

File: sirepo/util.py (hunk 2 above)
Comment (e-carlin) [reply sirepo/util.py]: Thanks. I like that better. 7a3d2ae8

File: sirepo/util.py (hunk 1 above)
Comment (e-carlin) [reply sirepo/util.py]: 7a3d2ae8

**Review by robnagler — APPROVED** [review]

**Pull request merged**